pkg/domain/infra/abi/pods.go

```diff
@@ -83,6 +83,46 @@ func (ic *ContainerEngine) PodKill(ctx context.Context, namesOrIds []string, opt
 	return reports, nil
 }
 
+func (ic *ContainerEngine) PodLogs(ctx context.Context, nameOrID string, options entities.PodLogsOptions) error {
+	// Implementation accepts slice
+	podName := []string{nameOrID}
+	pod, err := getPodsByContext(false, options.Latest, podName, ic.Libpod)
+	if err != nil {
+		return err
+	}
+	// Get pod containers
+	podCtrs, err := pod[0].AllContainers()
+	if err != nil {
+		return err
+	}
+
+	ctrNames := []string{}
+	// Check if `kubectl pod logs -c ctrname <podname>` alike command is used
+	if options.ContainerName != "" {
+		ctrFound := false
+		for _, ctr := range podCtrs {
+			if ctr.ID() == options.ContainerName || ctr.Name() == options.ContainerName {
+				ctrNames = append(ctrNames, options.ContainerName)
+				ctrFound = true
+			}
+		}
+		if !ctrFound {
+			return errors.Wrapf(define.ErrNoSuchCtr, "container %s is not in pod %s", options.ContainerName, nameOrID)
+		}
+	} else {
+		// No container name specified select all containers
+		for _, ctr := range podCtrs {
+			ctrNames = append(ctrNames, ctr.Name())
+		}
+	}
+
+	// PodLogsOptions are similar but contains few extra fields like ctrName
+	// So cast other values as is so we can re-use the code
+	containerLogsOpts := entities.PodLogsOptionsToContainerLogsOptions(options)
+
+	return ic.ContainerLogs(ctx, ctrNames, containerLogsOpts)
```

Comment:
What happens if a container is added to the pod - should we include the new container in the log output?

Reply:
@mheon There is  going to be this race condition if container is added between a ongoing `podman pod logs` user will not see logs of newly added container until he invokes command again.However on every new invoke of `podman pod logs` user should see logs of new container. 

I think `kubectl` would have this same drawback but pods are not updated there instead new pods are spun.

We can definitely use channels and keep updating container list dynamically but for sure that will add some complexity and cost. One question is that complexity worth implementing here. WDYT ?

Reply:
I don't think the extra complexity is worth it.

Reply:
Confirmed from @mheon we are adding a note to doc for this and not implementing this edge case for now.